dua_manager.cpp bad ifdefs around prefix and exit

**Describe the bug** A clear and concise description of what the bug is.

Compile error with the following command line

**To Reproduce** Information to reproduce the behavior, including:

1. Git commit id

5276 (I think)

2. IEEE 802.15.4 hardware platform

nrf52840dk

3. Build steps

make -f examples/Makefile-nrf52840 clean
make -f examples/Makefile-nrf52840 COMMISSIONER=1 JOINER=1 COAP=1 DNS_CLIENT=1 MTD_NETDIAG=1 BORDER_ROUTER=1 MAC_FILTER=1 UDP_PROXY=1 BORDER_AGENT=1 SNTP_CLIENT=1 THREAD_VERSION=1.2 TIME_SYNC=1

4. Network topology

**Expected behavior** A clear and concise description of what you expected to happen.

Clean compile

**Console/log output** If applicable, add console/log output to help explain your problem.

  CXX      thread/libopenthread_ftd_a-network_data_leader_ftd.o
/Users/riedlse/Documents/GitHub/openthread/examples/../src/core/thread/dua_manager.cpp: In member function 'void ot::DuaManager::HandleDomainPrefixUpdate(ot::BackboneRouter::Leader::DomainPrefixState)':
/Users/riedlse/Documents/GitHub/openthread/examples/../src/core/thread/dua_manager.cpp:92:24: error: unused variable 'prefix' [-Werror=unused-variable]
   92 |     const Ip6::Prefix *prefix = nullptr;
      |                        ^~~~~~
/Users/riedlse/Documents/GitHub/openthread/examples/../src/core/thread/dua_manager.cpp:150:1: error: label 'exit' defined but not used [-Werror=unused-label]
  150 | exit:
      | ^~~~
  CXX      thread/libopenthread_ftd_a-network_data_local.o



**Additional context** Add any other context about the problem here.

I think the fix is to move #endif from 148 to 151 and add 
#if OPENTHREAD_CONFIG_DUA_ENABLE / #endif around line 92, but not sure what the code is doing, so I don't want to commit.

--- a/src/core/thread/dua_manager.cpp
+++ b/src/core/thread/dua_manager.cpp
@@ -89,8 +89,6 @@ DuaManager::DuaManager(Instance &aInstance)
 
 void DuaManager::HandleDomainPrefixUpdate(BackboneRouter::Leader::DomainPrefixState aState)
 {
-    const Ip6::Prefix *prefix = nullptr;
-
     if ((aState == BackboneRouter::Leader::kDomainPrefixRemoved) ||
         (aState == BackboneRouter::Leader::kDomainPrefixRefreshed))
     {
@@ -123,16 +121,17 @@ void DuaManager::HandleDomainPrefixUpdate(BackboneRouter::Leader::DomainPrefixSt
         // fall through
     case BackboneRouter::Leader::kDomainPrefixRefreshed:
     case BackboneRouter::Leader::kDomainPrefixAdded:
-        prefix = Get<BackboneRouter::Leader>().GetDomainPrefix();
+    {
+        const Ip6::Prefix *prefix = Get<BackboneRouter::Leader>().GetDomainPrefix();
         OT_ASSERT(prefix != nullptr);
-        break;
+        mDomainUnicastAddress.mPrefixLength = prefix->GetLength();
+        mDomainUnicastAddress.GetAddress().Clear();
+        mDomainUnicastAddress.GetAddress().SetPrefix(*prefix);
+    }
+    break;
     default:
         ExitNow();
     }
-
-    mDomainUnicastAddress.mPrefixLength = prefix->GetLength();
-    mDomainUnicastAddress.GetAddress().Clear();
-    mDomainUnicastAddress.GetAddress().SetPrefix(*prefix);
 
     // Apply cached DUA Interface Identifier manually specified.
     if (IsFixedDuaInterfaceIdentifierSet())
@@ -145,10 +144,10 @@ void DuaManager::HandleDomainPrefixUpdate(BackboneRouter::Leader::DomainPrefixSt
     }
 
     AddDomainUnicastAddress();
-#endif
 
 exit:
     return;
+#endif
 }
 
 #if OPENTHREAD_CONFIG_DUA_ENABLE

